third_party/move/move-model/src/builder/model_builder.rs

```diff
@@ -467,6 +468,37 @@ impl<'env> ModelBuilder<'env> {
         self.const_table.insert(name, entry);
     }
 
+    /// Adds friend declarations for package visibility.
+    /// This should only be called when all modules are loaded.
+    pub fn add_friend_decl_for_package_visibility(&mut self) {
+        let target_modules = self
+            .env
+            .get_modules()
+            .filter(|module_env| module_env.is_target() && !module_env.is_script_module())
```

Comment:
Should be `is_primary_target`. Sometimes we have targets of compilation which are not part of the processed 'unit'.  The unit is actually the compiled package the way we use the compiler.

I forgot to mention this before, but the definition of the 'current package we are compiling' is exactly all `m` s.t. `m.is_primary_target()`. So we do not need this address or path magic. All package visibilities in non-is-target should be considered private. Can we update the PR this way?

Reply:
This method is called when the entire global environment has been built. So we should use `is_target`, and using `is_primary_target` will not properly add friend declarations for the dependency modules (tested). If we do this instead when building the module, then we can use `is_primary_target`.

The pipeline`function_checker` also uses `in_same_package` after the entire environment is built, so we still have to use the path magic.

Reply:
Nope. `is_target` is true also for modules outside of the package since a recent bug fix by Brian. is_primary_target is the right function to call. And if not, then we have to fix something there, but this code should be based on the assumption that `is_primary_target` is doing the right thing.

> will not properly add friend declarations for the dependency modules (tested)

In which setup is that happening? Unit test, transactional test? If so, then the test environment is broken. Please describe the problem instead of just overriding my request here.

We must not use this path hack. I'm not in favor of further regressing the architecture of model and compiler. This is a hard blocker for getting this PR landed.

Reply:
Switched to `is_primary_target`.